### PR TITLE
[dyld] Do iterate subfolders and other non-regular files.

### PR DESCRIPTION
**Pull request by vgvassilev**
This should improve root-project/root#7774

Kudos to Alexander Penev (@alexander-penev).

### CODE REVIEW

File: interpreter/cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp [hunk 1]
```diff
@@ -437,10 +437,16 @@ namespace cling {
...
              DirIt != DirEnd && !EC; DirIt.increment(EC)) {
 
+          if (DirIt->type() == llvm::sys::fs::file_type::directory_file)
+            continue;
+
           // FIXME: Use a StringRef here!
           std::string FileName = getRealPath(DirIt->path());
           assert(!llvm::sys::fs::is_symlink_file(FileName));
 
+          if (!llvm::sys::fs::is_regular_file(FileName))
```
Comment (Axel-Naumann) [inline interpreter/cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp:447]: We have (seemingly) the same check in `DynamicLibraryManager::isSharedLibrary()` as called from `ShouldPermanentlyIgnore()`. Should we keep both, or move everything to `DynamicLibraryManager::isSharedLibrary()`?

File: interpreter/cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp [hunk 2]
```diff
@@ -437,10 +437,16 @@ namespace cling {
         for (llvm::sys::fs::directory_iterator DirIt(DirPath, EC), DirEnd;
              DirIt != DirEnd && !EC; DirIt.increment(EC)) {
 
+          if (DirIt->type() == llvm::sys::fs::file_type::directory_file)
```
Comment (Axel-Naumann) [inline interpreter/cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp:440]: Also this check is already in `DynamicLibraryManager::isSharedLibrary()`

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by vgvassilev** [comment]
This PR is no longer needed as the underlying issue was addressed by PR #6969

**Pull request closed without merging**